msg/adc_report.msg

```diff
@@ -1,3 +1,6 @@
 uint64 timestamp		# time since system start (microseconds)
-int16[12] channel_id		# ADC channel IDs, negative for non-existent
-float32[12] channel_value	# ADC channel value in volt, valid if channel ID is positive
+uint32 device_id		# unique device ID for the sensor that does not change between power cycles
+int16[12] channel_id		# ADC channel IDs, negative for non-existent, should be kept same as array index
+int32[12] raw_data		# ADC channel raw value, accept negative value, valid if channel ID is positive
+uint32[12] resolution		# ADC channel resolution, valid if channel ID is positive
```

Comment:
resolution is the same for all channels, and can be uint16

Reply:
But [here](https://github.com/PX4/Firmware/blob/a3710fcdefa2eb109f5250908c7b2ebf0301c9fc/src/drivers/drv_adc.h#L93)?

Reply:
You can change it there as well, or keep everything uint32, but we don't need that much.

Reply:
An ADC with 16bit precision will have this value of 65536, which is overflowing the uint16_t.

Reply:
I thought we weren't at the limit yet. Alright then it's clear.

Reply:
> 
> 
> I thought we weren't at the limit yet. Alright then it's clear.

What? Isn't stm32h7 already using 16bit ADC? https://github.com/PX4/Firmware/blob/0d36e5094b4655c960846d481f62361d40cf1d9f/platforms/nuttx/src/px4/stm/stm32h7/adc/adc.cpp#L312-L315

Reply:
It is, so we go with uint32.